src/sentry/search/events/builder.py

```diff
@@ -1405,3 +1421,106 @@ def resolve_additional_groupby(self) -> List[SelectType]:
+            Condition(self.column("organization_id"), Op.EQ, self.params["organization_id"])
+        )
+        return conditions
+
+    def resolve_snql_function(
+        self,
+        snql_function: MetricsFunction,
+        arguments: Mapping[str, NormalizedArg],
+        alias: str,
+        resolve_only: bool,
+    ) -> Optional[SelectType]:
+        if snql_function.snql_distribution is not None:
+            resolved_function = snql_function.snql_distribution(arguments, alias)
+            if not resolve_only:
+                self.distributions.append(resolved_function)
+                # Still add to aggregates so groupby is correct
+                self.aggregates.append(resolved_function)
+            return resolved_function
+        if snql_function.snql_set is not None:
+            resolved_function = snql_function.snql_set(arguments, alias)
+            if not resolve_only:
+                self.sets.append(resolved_function)
+                # Still add to aggregates so groupby is correct
+                self.aggregates.append(resolved_function)
+            return resolved_function
+        if snql_function.snql_counter is not None:
+            resolved_function = snql_function.snql_counter(arguments, alias)
+            if not resolve_only:
+                self.counters.append(resolved_function)
+                # Still add to aggregates so groupby is correct
+                self.aggregates.append(resolved_function)
+            return resolved_function
+        return None
+
+    def _default_filter_converter(self, search_filter: SearchFilter) -> Optional[WhereType]:
+        name = search_filter.key.name
+        operator = search_filter.operator
+        value = search_filter.value.value
+
+        lhs = self.resolve_column(name)
+        # resolve_column will try to resolve this name with indexer, and if its a tag the Column will be tags[1]
+        is_tag = isinstance(lhs, Column) and lhs.subscriptable == "tags"
+        if is_tag:
+            value = indexer.resolve(value)
+
+        # timestamp{,.to_{hour,day}} need a datetime string
+        # last_seen needs an integer
+        if isinstance(value, datetime) and name not in TIMESTAMP_FIELDS:
+            value = int(to_timestamp(value)) * 1000
+
+        if name in TIMESTAMP_FIELDS:
+            if (
+                operator in ["<", "<="]
+                and value < self.params["start"]
+                or operator in [">", ">="]
+                and value > self.params["end"]
+            ):
+                raise InvalidSearchQuery(
+                    "Filter on timestamp is outside of the selected date range."
+                )
+
+        # Tags are never null, but promoted tags are columns and so can be null.
+        # To handle both cases, use `ifNull` to convert to an empty string and
+        # compare so we need to check for empty values.
+        if is_tag:
+            if operator not in ["IN", "NOT IN"] and not isinstance(value, int):
```

Comment:
If it is `IN` or `NOT IN`, does there need to be a check that we're dealing with an array of ints?

Reply:
Good catch, also realized the way i'm resolving tag values doesn't handle `IN` queries either :+1: fixed that, and moved this check to the value resolution instead